### PR TITLE
Generate SDK from published preview models

### PR DESCRIPTION
**Pull request by bcelenza**
*Issue #, if available:*
aws/aws-app-mesh-controller-for-k8s#75

*Description of changes:*

Changes the Go SDK generation to use the [published SDK models for preview](https://github.com/aws/aws-app-mesh-roadmap/pull/210). This will allow us to support the controller working with the preview environment long term.

One minor side effect was needing to delete the `Tags: nil` references throughout the resource managers, as we don't publish the tagging operations in the preview SDK (we don't yet support it in preview). If and when we want tag support in the controller, we can definitely add it to the preview SDK -- but for now it isn't necessary to explicitly send the nil values.

*Testing:*

Built a regular and preview version of the controller:

```
IMG=bcelenza/app-mesh-controller:latest make docker-build
APPMESH_PREVIEW=1 IMG=bcelenza/app-mesh-controller:latest-preview make docker-build
```
Pushed the resulting images to docker hub.

Deployed the regular image to my IAD cluster to talk to the prod app mesh endpoint

Deployed the preview version to my preview cluster in PDX.

Tested creating a mesh in both clusters and verified they were created against the appropriate app mesh endpoints.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: appmesh_models_override/setup.sh [hunk 1]
```diff
@@ -4,16 +4,33 @@ set -e
 
 mkdir -p ./vendor/github.com/aws
 
+SDK_MODEL_SOURCE=https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/sdk
 SDK_VENDOR_PATH=./vendor/github.com/aws/aws-sdk-go
+API_VERSION=2019-01-25
+API_PATH=$SDK_VENDOR_PATH/models/apis/appmesh/$API_VERSION
+SERVICE_NAME=$([ "$APPMESH_PREVIEW" == "1" ] && echo "appmesh-preview" || echo "appmesh" )
```
Comment (fawadkhaliq) [inline appmesh_models_override/setup.sh:11]: Does not really impact this CR as this was the case before as well. Will `appmesh` service endpoint support `appmesh-preview` source model at line#7 or we expect some features may not be supported?

**Comment by fawadkhaliq** [comment]
Thanks for adding this
/lgtm

File: appmesh_models_override/setup.sh (hunk 1 above)
Comment (bcelenza) [reply appmesh_models_override/setup.sh:11]: The App Mesh production endpoint will not support all features in preview, but preview will support all features from App Mesh production. In effect, the preview SDK long-term will be a superset of production.

For features not supported in production, the controller will either error with a 404 (API doesn't exist) or error with a 403 (access denied). We protect new features with feature flags until they're ready to go GA, so it's generally safe to vend the preview models for production use.

Comment (fawadkhaliq) [reply appmesh_models_override/setup.sh:11]: Perfect, makes sense. Thanks!

**Pull request merged**